src/shogun/machine/IterativeMachine.h

```diff
@@ -0,0 +1,127 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Shubham Shukla
+ */
+
+#ifndef _ITERATIVEMACHINE_H__
+#define _ITERATIVEMACHINE_H__
+
+#include <shogun/lib/config.h>
+
+#include <shogun/base/progress.h>
+#include <shogun/lib/SGVector.h>
+#include <shogun/lib/common.h>
+#include <shogun/machine/LinearMachine.h>
+
+namespace shogun
+{
+	class CFeatures;
+	class CLabels;
+
+	/** @brief Class IterativeLinearMachine implements an iterative model
+	 * whose training can be prematurely stopped, and in particular be
+	 * resumed, anytime.
+	 */
+	template <class T>
+	class CIterativeMachine : public T
+	{
+	public:
+		/** Default constructor */
+		CIterativeMachine() : T()
+		{
+			m_current_iteration = 0;
+			m_complete = false;
+
+			SG_ADD(
+			    &m_current_iteration, "current_iteration",
+			    "Current Iteration of training", MS_NOT_AVAILABLE);
+			SG_ADD(
+			    &m_max_iterations, "max_iterations",
+			    "Maximum number of Iterations", MS_AVAILABLE);
+			SG_ADD(
+			    &m_complete, "complete", "Convergence status",
+			    MS_NOT_AVAILABLE);
+		}
+
+		virtual ~CIterativeMachine()
+		{
+		}
+		/** Returns convergence status */
+		bool is_complete()
+		{
+			return m_complete;
+		}
+
+		/** Continue Training
+		  *
+		  * Only possible if a previous CMachine::train call was prematurely
+		  * stopped.
```

Comment:
remove docstring here so that the one of CMachine is used (and the CMachine one should be polished a bit, see my comment)